pkg/execution/state/redis_state/lua/updateMetadata.lua

```diff
@@ -6,10 +6,39 @@ local ctx      = ARGV[1]
 local debugger = ARGV[2]
 local die      = ARGV[3] -- disable immediate execution
 local rv       = ARGV[4] -- request version
+local spanid   = ARGV[5] -- spanID
+local startts  = ARGV[6] -- fn start timestamp
+local sysdelay = ARGV[7] -- system delay duration milli
+local sojdelay = ARGV[8] -- sojourn delay duration milli
+
+local function is_field_empty(field, emptyval)
+  local val = redis.call("HGET", keyMetadata, field)
+  return val == nil or val == emptyval
+end
 
 redis.call("HSET", keyMetadata, "ctx", ctx)
 redis.call("HSET", keyMetadata, "die", die)
 redis.call("HSET", keyMetadata, "debugger", debugger)
 redis.call("HSET", keyMetadata, "rv", rv)
 
+-- only update the spanID if the existing value is empty
+if is_field_empty("sid", "") then
+  redis.call("HSET", keyMetadata, "sid", spanid)
+end
+
+-- only update the timestamp if it's the default value
+if is_field_empty("startedAt", "0") then
+  redis.call("HSET", keyMetadata, "startedAt", startts)
+end
+
+-- only update the duration if it's the default value
+if is_field_empty("sysdelay", "-1") then
+  redis.call("HSET", keyMetadata, "sysdelay", sysdelay)
+end
+
+-- only update the duration if it's the default value
+if is_field_empty("sojdelay", "-1") then
+  redis.call("HSET", keyMetadata, "sojdelay", sojdelay)
+end
```

Comment:
update the lua script to set span related metadata for the function run.
these are only set when they're not using the default values, so update should only happen once, which is on the start of the function run.